### PR TITLE
fix: Adjusted search criteria for Web Api

### PR DESCRIPTION
**Pull request by ShaBloze**
## Description

Adjusting QueryService to handle if charge types is null.
Also added versioning to Search Criteria dto.

## References

* #1656

### CODE REVIEW

File: source/Contracts/Charge/SearchCriteriaV1Dto.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@
 // ReSharper disable once CheckNamespace - Type is shared so namespace is not determined by project structure/namespace
 namespace Energinet.Charges.Contracts.Charge
 {
-    public record SearchCriteriaDto(
+    public record SearchCriteriaV1Dto(
         string ChargeIdOrName,
         Guid? OwnerId,
```
Comment (prtandrup) [inline source/Contracts/Charge/SearchCriteriaV1Dto.cs:23]: Sure this is a GUID and not the market participant ID (GLN)?

Comment (prtandrup) [reply source/Contracts/Charge/SearchCriteriaV1Dto.cs:23]: Clarified. Still expected to be able to search by GLN, but communication to backend will be using the GUID to secure uniqueness.

File: source/Energinet.Charges.Libraries/source/Energinet.DataHub.Charges.Clients.Registration/documents/release-notes/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Energinet.DataHub.Charges.Clients.Registrations Release notes
 
+## Version 4.0.0
+
+Bumped due to Clients package change
```
Comment (HenrikSommer) [inline source/Energinet.Charges.Libraries/source/Energinet.DataHub.Charges.Clients.Registration/documents/release-notes/release-notes.md:5]: Please explain the breaking change, that lead to a new major version.

**Review by prtandrup — APPROVED** [review]

File: source/Contracts/Charge/SearchCriteriaV1Dto.cs [hunk 3]
```diff
@@ -18,7 +18,7 @@
 // ReSharper disable once CheckNamespace - Type is shared so namespace is not determined by project structure/namespace
 namespace Energinet.Charges.Contracts.Charge
 {
-    public record SearchCriteriaDto(
+    public record SearchCriteriaV1Dto(
```
Comment (HenrikSommer) [inline source/Contracts/Charge/SearchCriteriaV1Dto.cs:21]: Why change the name, when there is no V2? - we might never need a V2

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/QueryServices/ChargesQueryServiceTests.cs [hunk 4]
```diff
@@ -361,31 +448,31 @@ private static Charge GetValidChargeWithMultiplePeriods(string currentName)
...
+                return marketParticipant.Id;
 
             marketParticipant = new MarketParticipant(
                 id: Guid.NewGuid(),
                 actorId: Guid.NewGuid(),
                 b2CActorId: Guid.NewGuid(),
-                MarketParticipantOwnerId,
+                marketParticipantOwnerId,
                 MarketParticipantStatus.Active,
                 MarketParticipantRole.EnergySupplier);
```
Comment (prtandrup) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/WebApi/QueryServices/ChargesQueryServiceTests.cs:471]: nit: EnergySuppliers do not create any charges, please use GridAccessProvider or SystemOperator

**Pull request merged**